libsysbox/syscont/spec.go

```diff
@@ -592,6 +607,20 @@ func cfgSysboxFsMounts(spec *specs.Spec, sysFs *sysbox.Fs) {
 			)
 	}
 
+	// If the spec indicates a read-only rootfs, the sysbox-fs mounts should also
+	// be read-only. However, we don't mark them read-only here explicitly, so
+	// that they are initially mounted read-write while setting up the container.
+	// This is needed because the setup process may need to write to some of
+	// these mounts (e.g., writes to /proc/sys during networking setup). Instead,
+	// we add the mounts to the "readonly" paths list, so that they will be
+	// remounted to read-only after the container setup completes, right before
+	// starting the container's init process.
+	if spec.Root.Readonly {
```

Comment:
Sounds like a good idea ... as long as this field isn't utilize in the future for something else in upstream runc and we miss it while syncing up ...

Reply:
This field is part of the OCI runtime spec, so it's meaning is well defined.